### PR TITLE
Merges Testing

### PR DESCRIPTION
**Pull request by navanchauhan**
Merges QOL improvements made by @ericgaspar and fixes upgrade script along with bumping version to ynh3

## Package_check results
---
* An automatic package_check will be launch at https://ci-apps-dev.yunohost.org/, when you add a specific comment to your Pull Request: "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!"*

### CODE REVIEW

**Comment by navanchauhan** [comment]
!testme

**Review by ericgaspar — COMMENTED** [review]
We don't need to change the version here. We just need to increment the Master +1 version so that it is `0.75.1~ynh2`

**Comment by navanchauhan** [comment]
I am incrementing the version to be able to test on my pi by using the upgrade command. This way I can ensure that the upgrade functionality works. I'll decrement it to ynh2 before merging.

**Comment by ericgaspar** [comment]
Just use `--force` command to  force the upgrade:
`sudo yunohost app upgrade komga -u https://github.com/YunoHost-Apps/komga_ynh/tree/testing --force`

**Comment by navanchauhan** [comment]
ahhhhhh, it is that simple 🤦‍♂️

**Comment by ericgaspar** [comment]
This app should install. For some rather curious reason, because of my YunoHost account registered with accentuated name, the app was gives me an error 500 ... When switching to an account where the registered name is not accented, the problem goes away.

**Comment by ericgaspar** [comment]
!testme

**Comment by navanchauhan** [comment]
> This app should install. For some rather curious reason, because of my YunoHost account registered with accentuated name, the app was gives me an error 500 ... When switching to an account where the registered name is not accented, the problem goes away.


That is indeed weird. Are you able to login to komga when you are logged into YunoHost or is issue #5 just happening to me?

**Comment by ericgaspar** [comment]
> > This app should install. For some rather curious reason, because of my YunoHost account registered with accentuated name, the app was gives me an error 500 ... When switching to an account where the registered name is not accented, the problem goes away.
> 
> That is indeed weird. Are you able to login to komga when you are logged into YunoHost or is issue #5 just happening to me?

I did get the same issue with Firefox but the registration process went through with Chromium. I don't know what the problem is ...

**Comment by navanchauhan** [comment]
> > > This app should install. For some rather curious reason, because of my YunoHost account registered with accentuated name, the app was gives me an error 500 ... When switching to an account where the registered name is not accented, the problem goes away.
> > 
> > 
> > That is indeed weird. Are you able to login to komga when you are logged into YunoHost or is issue #5 just happening to me?
> 
> I did get the same issue with Firefox but the registration process went through with Chromium. I don't know what the problem is ...

Are we passing any basic auth headers for logged in users? I disabled SSOwat injection on my local installation yet the problem still persists. The weird thing is that its working flawlessly for logged out use-cases (from registrations to logging in )

**Comment by ericgaspar** [comment]
You are right, for me it worked with Chromium because I wasn't logged in... maybe ping @kay0u who knows permissions better

File: conf/systemd.service [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 Type=simple
 User=__APP__
 Group=__APP__
 WorkingDirectory=__FINALPATH__/
-ExecStart=java -jar __FINALPATH__/app.jar --server.port=__PORT__ --server.servlet.context-path="__PATH__" >> /var/log/__APP__/__APP__.log 2>&1
+ExecStart=/usr/bin/java -jar -Xmx1g app.jar --server.port=__PORT__ --server.servlet.context-path="__PATH__/"
+Restart=on-failure
+RestartSec=10
+StandardOutput=null
+StandardError=syslog
```
Comment (kay0u) [inline conf/systemd.service:14]: @ericgaspar Uuuh, why?

Comment (ericgaspar) [reply conf/systemd.service:14]: @kay0u I followed the config proposed in this thread ... https://github.com/gotson/komga/issues/353

**Comment by kay0u** [comment]
https://github.com/YunoHost-Apps/komga_ynh/pull/7 should fix the issue

File: conf/systemd.service (hunk 1 above)
Comment (kay0u) [reply conf/systemd.service:14]: Ok, the `line_match` argument can't work if we discard all logs. I fixed it here: https://github.com/YunoHost-Apps/komga_ynh/pull/6

**Comment by navanchauhan** [comment]
Thanks @kay0u merging them

File: scripts/upgrade [hunk 2]
```diff
@@ -12,35 +12,32 @@ source /usr/share/yunohost/helpers
...
-### UPGRADE_APP should be used to upgrade the core app only if there's an upgrade to do.
 upgrade_type=$(ynh_check_app_version_changed)
 
 #=================================================
 # ENSURE DOWNWARD COMPATIBILITY
 #=================================================
-ynh_script_progression --message="Ensuring downward compatibility..." --time --weight=1
+ynh_script_progression --message="Ensuring downward compatibility..." --weight=1
+
+ynh_permission_url --permission="main" --auth_header="false"
```
Comment (kay0u) [inline scripts/upgrade:35]:
```suggestion
```

If we consider this app not used/installed yet, we can remove this line

Comment (navanchauhan) [reply scripts/upgrade:35]: Yup, I have not even announced this on the forum yet

**Comment by ericgaspar** [comment]
I installed komga on my server without problems so far. Thanks @ kay0u for the help

**Comment by navanchauhan** [comment]
> 😜
> [Test Badge](https://ci-apps-dev.yunohost.org/ci/job/537)

@ericgaspar the tests are failing :/ It is throwing 502 bad gateway which I assume is because it is not able to fully launch before the test checks if the url is accessible

```
The service komga didn't fully executed the action start before the timeout.
```

Can we change the timeout duration?

**Comment by navanchauhan** [comment]
I think we can merge this into master because everything is working even though the automated tests are failing

**Comment by ericgaspar** [comment]
You could use `--timeout=60` : https://github.com/YunoHost-Apps/etherpad_mypads_ynh/blob/8c8f0c9aff78c6a376a581d4323497135b05900a/scripts/install#L331
or `sleep 60`

**Comment by ericgaspar** [comment]
It also means  that `--line_match` option in not finding anything that matches

**Comment by navanchauhan** [comment]
!testme

**Comment by navanchauhan** [comment]
!testme

**Comment by navanchauhan** [comment]
Merging PR, tests are still failing though :/

Will create new PR for tests

**Pull request merged**